fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/entity/FabricEntityTypeBuilder.java

```diff
@@ -148,6 +154,27 @@ protected FabricEntityTypeBuilder(SpawnGroup spawnGroup, EntityType.EntityFactor
 		return this;
 	}
 
+	/**
+	 * Sets the default attributes for a type of living entity.
+	 *
+	 * <p>This should not be called if your entity is not a {@link LivingEntity}.
+	 *
+	 * <p>This can be used in a fashion similar to this:
+	 * <blockquote><pre>
+	 * FabricEntityTypeBuilder.create(SpawnGroup.CREATURE, MyCreature::new)
+	 * 	.defaultAttributes(LivingEntity::createLivingAttributes)
+	 * 	...
+	 * 	.build();
+	 * </pre></blockquote>
+	 *
+	 * @param defaultAttributeBuilder a function to generate the default attribute builder from the entity type
+	 * @return this builder for chaining
+	 */
+	public FabricEntityTypeBuilder<T> defaultAttributes(Supplier<DefaultAttributeContainer.Builder> defaultAttributeBuilder) {
+		this.defaultAttributeBuilder = defaultAttributeBuilder;
```

Comment:
We really need some mechanism to properly handle registration of living vs non living entities.

Reply:
Currently I don't register if they don't call this method.

The only ways I see being able to test this is creating a dummy world, creating a dummy entity and doing an instanceof or a bunch of reflection